### PR TITLE
Implement RejectedExecutionException.

### PR DESCRIPTION
**Pull request by andreaTP**
as always, just avoid linking errors.

### CODE REVIEW

File: javalib/src/main/scala/java/util/concurrent/Throwables.scala [hunk 1]
```diff
@@ -17,3 +17,13 @@ class CancellationException(message: String)
 class TimeoutException(message: String) extends Exception(message) {
   def this() = this(null)
 }
+
+class RejectedExecutionException(message: String, cause: Throwable)
+    extends RuntimeException(message) {
```
Comment (sjrd) [inline javalib/src/main/scala/java/util/concurrent/Throwables.scala]: `, cause`?

**Comment by andreaTP** [comment]
Akka ... ActorSystem throws it

**Comment by nicolasstucki** [comment]
I believe @sjrd ment the `cause` parameter ;)

**Comment by andreaTP** [comment]
XD ... ok ...
just copy paste from:
https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/RejectedExecutionException.html#RejectedExecutionException(java.lang.Throwable)

**Comment by nicolasstucki** [comment]
Just don't forget pass it to the `extends RuntimeException(message, cause)`.

**Comment by andreaTP** [comment]
argh... yep doing it!

**Comment by andreaTP** [comment]
done! Tnx!

File: javalib/src/main/scala/java/util/concurrent/Throwables.scala [hunk 2]
```diff
@@ -17,3 +17,13 @@ class CancellationException(message: String)
...
+  
+  def this() = this(null, null)
+
+  def this(message: String) = this(message, null)
+
+  def this(cause: Throwable) = this(null, cause)
```
Comment (nicolasstucki) [inline javalib/src/main/scala/java/util/concurrent/Throwables.scala]: The doc says that the message should be ` (cause == null ? null : cause.toString())`

**Comment by nicolasstucki** [comment]
That is all

**Comment by andreaTP** [comment]
please double check the style targeting your comment.

File: javalib/src/main/scala/java/util/concurrent/Throwables.scala [hunk 3]
```diff
@@ -17,3 +17,18 @@ class CancellationException(message: String)
...
+  def this() = this(null, null)
+
+  def this(message: String) = this(message, null)
+
+  def this(cause: Throwable) =
+    this( 
+      {if (cause eq null) null
+      else cause.toString},
+      cause
+    )
```
Comment (nicolasstucki) [inline javalib/src/main/scala/java/util/concurrent/Throwables.scala]: Do it in one line.
```scala
this(if (cause eq null) null else cause.toString, cause)
```

**Comment by andreaTP** [comment]
done

**Comment by nicolasstucki** [comment]
LGTM

**Comment by nicolasstucki** [comment]
Jenkins, test this please

**Comment by sjrd** [comment]
Note: No merge at the moment because we're in the release process.

File: javalib/src/main/scala/java/util/concurrent/Throwables.scala [hunk 4]
```diff
@@ -17,3 +17,14 @@ class CancellationException(message: String)
 class TimeoutException(message: String) extends Exception(message) {
   def this() = this(null)
 }
+
+class RejectedExecutionException(message: String, cause: Throwable)
+    extends RuntimeException(message, cause) {
+
```
Comment (sjrd) [inline javalib/src/main/scala/java/util/concurrent/Throwables.scala]: Whitespace at end-of-line, says CI.

**Comment by andreaTP** [comment]
fixed... sorry.. forgot to run scalastyle

**Comment by sjrd** [comment]
This can now be rebased on top of master.

**Comment by andreaTP** [comment]
here you are, just to know can I expect a 6.10 release before scala days berlin?

**Comment by gzm0** [comment]
@sjrd I think it might be sensible to roll a bugfix release pretty soon (end of May?). Quite some things lined up in 0.6.10 already. WDYT?

**Comment by sjrd** [comment]
LGTM, Jenkins, test this please

**Comment by sjrd** [comment]
@gzm0 I would aim for ~ June 4-7 period, one week before Scala Days Berlin.

**Pull request merged**